File: src/001/p001.c
#include <stdio.h>

int solve() {
    int i, n = 0;
    for (i = 1; i < 1000; i++) {
        if (i % 3 == 0 || i % 5 == 0) {
            n += i;
        }
    }
    return n;
}

int main() {
    int result = solve();
    printf("%d\n", result);
}

File: src/001/p001.ml
open Base
open Stdio

let solve =
  List.range 1 1000
  |> List.filter ~f:(fun x -> x % 3 = 0 || x % 5 = 0)
  |> List.reduce ~f:(+)

let () =
  printf "%d\n" solve;;
